### PR TITLE
ncm-authconfig: fix krb5_realm, user_object_class and newline between ldap_krb5_canonicalize and ldap_krb5_realm

### LINKED ISSUE #1100: ncm-authconfig : no newline after krb5_canonicalize option
When I do an ncm-ncd --configure authconfig, the /etc/sssd.conf file contains the following line:
Krb5_canonicalize = Falsekrb5_realm = LAPP.IN2P3.EN
While it should be:
Krb5_canonicalize = False
Krb5_realm = LAPP.IN2P3.EN

This is due to an error in the file /usr/share/templates/quattor/authconfig/domains/ldap.tt

[% - SWITCH name -%]
[% - CASE ['krb5_canonicalize'] -%]
[% - desc. $ Name? 'True': 'False'%]
**[% - CASE -%]**
[% - desc. $ Name%]
[% END -%]

  should be

[% - SWITCH name -%]
[% - CASE ['krb5_canonicalize'] -%]
[% - desc. $ Name? 'True': 'False'%]
**[% CASE -%]**
[% - desc. $ Name%]
[% END -%]

### LINKED ISSUE #1103: ncm-authconfig: krb5_realm = ARRAY(0x39a5830)
When I deploy my sssd config with the option `"krb5_realm", list(LAPP.IN2P3.fr)"`
I obtain this error in `/etc/sssd/sssd.conf`:
`krb5_realm = ARRAY(0x39a5830)`

In the authconfig schema, if I replace `"krb5_realm" ? string[]` by `"krb5_realm" ? string`, all its ok.

### PR DESCRIPTION
**Pull request by jrha**
- Add tests for Issues #1100 #1102 and #1103
- Make "krb5_realm" option as a single string add remove "use_object_class" option in sssd/ldap.pan schema
- Fix ldap.tt for insert newline between "ldap_krb5_canonicalize" and "ldap_krb5_realm" in the ssd.conf.file

Fixes #1100, fixes #1103
Replaces #1105, @seraf1's original PR which somehow became orphaned.

Breaks backwards compatibility by changing the type of `krb5_realm` from a list of strings to simply a string.

### CODE REVIEW

**Comment by ned21** [comment]
@ajf8 - anything here which might interfere with the change you are working on at the moment?

File: ncm-authconfig/src/main/resources/tests/profiles/ldap.pan [hunk 1]
```diff
@@ -15,5 +15,7 @@ prefix "/desc";
 "tls/cacert" = 'ca';
 "tls/cipher_suite" = list("c1", "c2");
 "user/shell" = "us";
-"uri" = list("u1","u2");
-
+"uri" = list("u1", "u2");
+"user/object_class" = "user"
```
Comment (ned21) [inline ncm-authconfig/src/main/resources/tests/profiles/ldap.pan]: Missing ; causing test failure

**Comment by jrha** [comment]
@hpcugentbot retest this please

**Comment by jrha** [comment]
The type change seems correct to me, as the man page states:
> krb5_realm (string)
>     The name of the Kerberos realm. This option is required and must be specified.

**Comment by jrha** [comment]
Once again tests failing at the JUnit step. @stdweird is something wrong with Jenkins?

**Comment by ned21** [comment]
The commit log in dad7cd2 does not match the change. It would be good to squash 6ec2c02 into the first commit too since they are touching the same area.

The rest looks fine.

**Comment by jrha** [comment]
@ned21 thanks! Rebased, squashed and reworded.

**Comment by ned21** [comment]
The first line of a35a454 still seems a but garbled?

```
ncm-authconfig: make krb5_realm option as a single string add remove use_object_class option

krb5_realm as string (and not array of string) -> Fixes #1102
```

**Comment by jrha** [comment]
Whoops, thanks for catching that.

**Pull request merged**